src/app/api/common/delegations/getDelegations.ts

```diff
@@ -21,86 +22,55 @@ async function getCurrentDelegateesForAddress({
 }): Promise<Delegation[]> {
   const { namespace, contracts } = Tenant.current();
 
-  const advancedDelegatees = await prisma[
-    `${namespace}AdvancedDelegatees`
-  ].findMany({
-    where: {
-      from: address.toLowerCase(),
-      delegated_amount: { gt: 0 },
-      contract: contracts.alligator?.address,
-    },
-  });
-
-  // const directDelegatee = await (async () => {
-  //   const [proxyAddress, delegatee] = await Promise.all([
-  //     getProxyAddress(address),
-  //     prisma.delegatees.findFirst({
-  //       where: { delegator: address.toLowerCase() },
-  //     }),
-  //   ]);
+  const [advancedDelegatees, directDelegatee] = await Promise.all([
```

Comment:
wonder if this would benefit from prisma transaction 
https://www.prisma.io/docs/orm/prisma-client/queries/transactions

although the second promise being another nested promise is confusing me a bit

Reply:
ah prisma transactions are great when you want yo ensure that queries are executed atomically meaning that if one fails, they all fail. This is great when you're making updates to the DB.
Here, I just want to parallelize the query execution. Also, Prisma's execution engine tends to prioritize safety over speed and in read only operations speed is all that matters. So I'd rather take this control away from prisma.

The readability is not the greatest here, you're right -- will refactor